boltops-tools/jets
# Dont bypass class_attribute setters

This is a 🐞 bug fix.






## Summary

The skip_before_action method no longer modifies class attributes directly but now uses the setters

## Context

https://github.com/boltops-tools/jets/issues/480

## How to Test

1. add a before_action callback to the ApplicationController class
2. create a controller derived from ApplicationController
3. create a second controller derived from ApplicationController and use the skip_before_action with the only option to skip the callback created in ApplicationController for one or more methods

Notice, The first controller should still execute the before_action on all methods.

```
class ApplicationController < Jets::Controller::Base
  before_action :before_callback

  def before_callback
    puts "ApplicationController::before called"
  end
end

class TestAController < ApplicationController
  def index
    puts "TestAController::index called"
  end
end

class TestBController < ApplicationController
  skip_before_action :before_callback, only: %i[index]

  def index
    puts "TestBController::index called"
  end
end

```
